docassemble/ALKilnInThePlayground/data/questions/run_alkiln_tests.yml

```
@@ -2159,83 +2552,428 @@ content: |
+  Return the runtime_config.json path or, if it doesn't yet exist, None
+  """
+  # temp_config_path = get_existing_path( "runtime_config.json" )
+  # if temp_config_path is None:
+  #   config_path_problem = 'No "runtime_config.json" file exists.'
+  #   _log(f'🤕 ALKiln ERROR: { config_path_problem }')
+  # config_path = temp_config_path
+  config_path = get_config_path()["result"]
 ---
 need:
   - get_existing_path
 code: |
-  temp_config_path = get_existing_path( "runtime_config.json" )
-  if temp_config_path is None:
-    config_path_problem = 'No "runtime_config.json" file exists.'
-    _log(f'🤕 ALKiln ERROR: { config_path_problem }')
-  config_path = temp_config_path
+  def get_config_path():
+    """
+    Return the runtime_config.json path or, if it doesn't yet exist, None
+    """
+    ok = True
+    message = ""
+    result = get_existing_path( "runtime_config.json" )
+  
+    if result is None:
+      ok = False
+      message = 'No "runtime_config.json" file exists.'
+      _log(f'🔎 ALKiln WARNING: { message }')
+    return { "ok": ok, "message": message, "result": result }
 ---
 id: default config_path_problem
 code: |
-  config_path_problem = ''
+  config_path_problem = ''  # Default
 ---
 # We get the folder_path by using ALKiln's runtime_config.json
 # folder_path tries different locations till it finds the real one in order to stay compatibile with old ALKiln versions. See `get_existing_path()` notes. We use `folder_path` to get the artifacts folder to make zip names, zips, and to eventually delete the folder.
-id: artifacts folder name
+id: artifacts paths and names
 need:
   - config_path
   - get_existing_path
 code: |
   import os
+  import traceback
+
+  # _debug(f"""folder_name trace: {"\n".join( traceback.print_stack() )}""")
   
   if config_path is None:
     folder_temp_name = 'no config'
 
   else:
-    with open( config_path ) as config_text:
+    with open( config_path ) as config_file:
 
       try:
-        folder_temp_name = json.load( config_text ).get('artifacts_path', 'no value')
+        folder_temp_name = json.load( config_file ).get('artifacts_path', 'no json value')
   
       except Exception as config_load_error:
-        folder_temp_name = 'no json'
-        _log(f'🤕 ALKiln ERROR: Error loading json from config file at "{ config_path }":')
-        _log( config_load_error )
+        folder_temp_name = 'json load error'
+        folder_path_problem = f"""Unable to load JSON from the ALKiln runtime_config.json file at "{ config_path }":\n{"\n".join( traceback.format_exception( config_load_error ) )}"""
+        _log(f'🤕 ALKiln ERROR ALKP0017: { folder_path_problem }')
+        _log( config_file )
   
-      if folder_temp_name == 'no value':
-        _log(f'🤕 ALKiln ERROR: Config file at "{ config_path }" has no "artifacts_path" value.')
+    if folder_temp_name == 'no json value':
+      folder_path_problem = """The ALKiln runtime_config.json file at "{ config_path }" has no "artifacts_path" value."""
```

If we keep this around, we'd want to remove previous logs and move this into the `if` below with a fallback message. I suggest we log in-place, though. The traceback will make more sense when we add it.

```suggestion
    if folder_path_problem != "":
```